scripts/gendocs/generate

```diff
@@ -83,6 +100,54 @@ gendocs() {
 	rm -rf vendor/
 	popd
 }
+
+# genclireference will attempt to run main.go --help for the target and write the output to a markdown file
+genclireference() {
+  if [ ! -f "$2/main.go" ]; then
+    echo "+++ target $2 does not exist, skipping..."
+    return
+  fi
+
+  # hacky way to figure out if the target has the correct structure
+  # differs between older version but this catches the corner cases
+  if [[ ! -d "$2/app" ]] && [ ! -d "$2/cmd" ]; then
+    echo "+++ app directory for $2 does not exist, skipping..."
+    return
+  fi
+
+  # combined with the check above we can handle all versions
+  # for example release-0.15 webhook has the correct directory structure but does not use cobra
+  if ! grep -q "@com_github_spf13_cobra//:go_default_library" "$2/app/BUILD.bazel" && ! grep -q "@com_github_spf13_cobra//:go_default_library" "$2/cmd/BUILD.bazel"; then
+    return
+  fi
+
+  outputdir="$1"
+  target="$2"
+  name="$3"
+  echo "+++ Generating CLI reference docs for $target ..."
+
+  mkdir -p "${REPO_ROOT}/content/en/${outputdir}/cli/"
+
+  output=$(go run "$target/main.go" --help)
```

Comment:
question: was continuing the work of generating the documentation via cobra considered? https://github.com/jetstack/cert-manager/blob/master/tools/cobra/main.go

Not a huge thing, but it does  give a _sightlier_ better output format, and can be configured with headers etc.

Reply:
Going forward it could definitely make sense to build on top of that, the new website will not be based on Hugo so it could be we need something purpose-built rather than this simple script.

At this stage we decided to simply use the output of `--help`, and we wanted the CLI reference for older versions.

Reply:
Tested it a bit now and it's not a huge change to use the tool you linked for the versions where it's available, but it requires some work on both ends and I'm not sure it's worth it?

Reply:
Thought: I'm surprised that --help doesn't output to standard error.